scripts/uberenv/packages/axom/package.py

```diff
@@ -524,12 +536,17 @@ def cmake_args(self):
 
         options = []
         options.extend(['-C', host_config_path])
+
         if self.run_tests is False:
             options.append('-DENABLE_TESTS=OFF')
         else:
             options.append('-DENABLE_TESTS=ON')
-        src_dir = os.path.abspath(pjoin(self.stage.source_path, "src"))
-        options.append(src_dir)
+
+        if "+shared" in spec:
+            options.append('-DBUILD_SHARED_LIBS=ON')
+        else:
+            options.append('-DBUILD_SHARED_LIBS=OFF')
+
```

Comment:
Should we consider putting all the CMake arguments in the host-config file?

FYI, I am asking because this is what I had to do on Umpire/RAJA/CHAI, so I made the host-config file name unique using the spack spec hash. 
Pros:
- Reusing the host-config file then leads to the exact same build.
- All the CMake parameters are handle at the same level in the Spack package.

Off-topic though...

Reply:
So i generally put things in the host-config.. but I felt this one leaned more towards the individual build option then needing to be in there for reproducibility.